### PR TITLE
3540: Added integration tests for password reset and user account confirmation

### PR DESCRIPTION
**Pull request by CallumBradburyBJSS**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    When a POST request is made to create a user for organisation Organisation 1
+        | FirstName | LastName   | PhoneNumber | EmailAddress             | OrganisationName |
+        | Bob       | Bobkovitch | 0123456789  | bob.bobkovitch@email.com | Organisation 1   |
+    Then a response with status code 201 is returned
+    And the email sent contains the following information
+        | From                           | To                       | Subject                                            |
+        | noreply@buyingcatalogue.nhs.uk | bob.bobkovitch@email.com | INTEGRATION_TEST Set password for Buying Catalogue |
+    And only one email is sent
+
+@3540
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:26]: What is 3540? Should it not be 3926?

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    When the user navigates to a restricted web page
+    Then the user is redirected to page identity/account/login
+    When a login request is made with email address bob.bobkovitch@email.com and password Password123!
+    Then the user is redirected to page identity/Consent
+    When element with Data ID agree-terms-checkbox is clicked
+    And element with Data ID agreement-submit-button is clicked
+    Then the user is redirected to page home/privacy
+    And the page contains element with ID sampleResourceResult with text Authorized With Sample Resource
+
+@3540
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:43]: Havent you already done this test in "Reset Password.3926"? Is there any need to do it again

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    When element with Data ID input-reset-password is populated with Password123!
+    And element with Data ID input-confirm-reset-password is populated with Password123!
+    And element with Data ID reset-password-button is clicked
+    Then the user is redirected to page account/resetpasswordconfirmation
+    When The email link is clicked
+    Then the user is redirected to page account/resetpasswordexpired
+
+@3540
+Scenario: 3. User cannot modify the link to set someone elses password
+    When The email link is modified to use email address PostmanPat@email.com
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:56]: Again with this test, duplication is shown in "Reset Password.3926"

Comment (CallumBradburyBJSS) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:56]: potentially, but it's two different routes, one is password reset one is user created, so I think it's good to check it for both scenarios even though under the hood it's using the same logic

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature (hunk 1 above)
Comment (CallumBradburyBJSS) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:26]: 3540 is new user created & email sent, which is where these tests belong. 3926 is password reset via forgot password link

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature (hunk 2 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:43]: After our call, you're right, better to test both scenarios 😄

File: tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature (hunk 3 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/Features/Set New User Password.3540.feature:56]: After our call, you're right, better to test both scenarios 😄

**Pull request merged**